src/test/scala/io/iohk/ethereum/consensus/ethash/RestrictedEthashSignerSpec.scala

```diff
@@ -0,0 +1,45 @@
+package io.iohk.ethereum.consensus.ethash
+
+import akka.util.ByteString
+import io.iohk.ethereum.ObjectGenerators
+import io.iohk.ethereum.nodebuilder.SecureRandomBuilder
+import org.bouncycastle.crypto.params.ECPublicKeyParameters
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+
+class RestrictedEthashSignerSpec
+    extends AnyFlatSpec
+    with Matchers
+    with ScalaCheckPropertyChecks
+    with ObjectGenerators
+    with SecureRandomBuilder {
+  "RestrictedEthashSigner" should "sign and validate correct header" in {
+    forAll(blockHeaderGen, genKey(secureRandom)) { (header, key) =>
```

Comment:
should similar tests be written against `RestrictedEthashHeaderValidator`?

Reply:
tbh i am not sure that is possible (or desirable) as `RestrictedEthashBlockHeaderValidator` inherits from `EthashBlockHeaderValidator` which mean method `validate` validates proof of work therefore to property test it one should mine valid blocks which would take too much time.

Thats why I had split that way:
- property test for signing and validating to test various different inputs 
- `RestrictedEthashBlockHeaderValidatorSpec` which tests different `if-else` parts from `RestrictedEthashBlockHeaderValidator` plus the case which test against someone who would try to re-sign header after pow is mined

Reply:
On the BlockHeaderValidatorSpec we used:
```
class BlockValidatorWithPowMocked(blockchainConfig: BlockchainConfig)
      extends BlockHeaderValidatorSkeleton(blockchainConfig) {
    override protected def difficulty: DifficultyCalculator = difficultyCalculator

    override def validateEvenMore(
        blockHeader: BlockHeader,
        parentHeader: BlockHeader
    ): Either[BlockHeaderError, BlockHeaderValid] = Right(BlockHeaderValid)
  }
```
Something like that might be useful for what you are saying